sklearn/linear_model/_ridge.py

```diff
@@ -2212,16 +2264,20 @@ def fit(self, X, y, sample_weight=None):
             parameters = {"alpha": alphas}
             solver = "sparse_cg" if sparse.issparse(X) else "auto"
             model = RidgeClassifier if is_classifier(self) else Ridge
+            est = model(
+                fit_intercept=self.fit_intercept,
+                solver=solver,
+            )
+            if _routing_enabled():
+                est.set_fit_request(sample_weight=True)
+
             gs = GridSearchCV(
-                model(
-                    fit_intercept=self.fit_intercept,
-                    solver=solver,
-                ),
+                est,
                 parameters,
                 cv=cv,
                 scoring=self.scoring,
             )
-            gs.fit(X, y, sample_weight=sample_weight)
+            gs.fit(X, y, sample_weight=sample_weight, **params)
```

Comment:
we should pack sample_weight in `params` and only include `sample_weight` if it's not None.